<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- SETUP ARGUMENTS -->
    <!-- phobos / phobos_nuclear -->
    <arg name="camera_model" default="phobos_nuclear"/>
    <!-- enable/disable rviz -->
    <arg name="rviz" default="false" />
    <!-- is camera with split laser functionality being used -->
    <arg name="split_laser" default="false"/>
    <!-- 0=TIS, 1=Basler -->
    <arg name="camera_type" default="1"/>
    <!-- namespace for camera -->
    <arg name="camera_namespace" default="$(arg camera_model)"/>

    <!-- Use full res camera model mesh (slower) -->
    <arg name="camera_mesh" default="false" />

    <arg name="urdf_namespace" default="$(arg camera_namespace)"/>

    <arg name="imu" default="false"/>
    <arg name="publish_imu_tf" default="false"/>
    <arg name="gui" default="false" />

    <param name="use_sim_time" value="true"/>

    <!--
    <param name="$(arg camera_model)_robot_description" command="$(find xacro)/xacro '$(find i3dr_phobos)/urdf/i3dr_phobos_scene.urdf.xacro' en_imu:=$(arg imu) camera_name:=$(arg urdf_namespace) camera_model:=$(arg camera_model) en_mesh:=$(arg camera_mesh)" />
    <node name="robot_state_publisher_phobos_nuclear" pkg="robot_state_publisher" type="state_publisher">
        <remap from="robot_description" to="$(arg camera_model)_robot_description"/>
    </node>
    -->

    <arg name="bag_file" value="'/home/i3dr/i3dr_mapping/data/i3dr_wood_scan/i3dr_ROS_room_scan_2019-07-04-15-45-00[Light on - Laser on].bag'"/>
    <arg name="bag_rate" value="2" />

    <!-- load capture from bag -->
    <node pkg="rosbag" type="play" name="player" output="screen" args="--clock -r $(arg bag_rate) $(arg bag_file)"/>

    <node pkg="i3dr_stereo_camera" type="rectify" ns="$(arg camera_namespace)" name="$(anon rectify)" output="screen">
    </node>
    
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="$(arg camera_model)_capture_rivz" 
        args="-d $(find i3dr_phobos)/rviz/$(arg camera_model)_standalone_scene.rviz"/>

    <node if="$(arg gui)" pkg="i3dr_stereo_camera" type="stereo_gui" name="$(anon stereo_gui)" ns="$(arg camera_namespace)" output="screen"/>

</launch>
